### ISSUE TITLE
关于maml.py中第149行

### ISSUE THREAD

**Issue by Gaotttt** [RobinLu1209/ST-GFSL#3]
作者您好，有些元学习训练时的疑惑，采用deepcopy不会导致优化器没办法优化嘛

**Comment by canyangguo** [comment]
> 作者您好，有些元学习训练时的疑惑，采用deepcopy不会导致优化器没办法优化嘛

我也有这个疑问，deeocopy的话，self.model的参数就变了，就不是θ，而是θ‘了，这和文章中伪代码的Line 11 有出入。
另外，我发现表2 (M2): Remove parameter generator的实验结果和表1的MAML一样，难道MAML==(M2): Remove parameter generator，据我所知，MAML应该是没有graph reconstruction loss的吧

**Comment by HQ-LV** [comment]
> > 作者您好，有些元学习训练时的疑惑，采用deepcopy不会导致优化器没办法优化嘛
> 
> 我也有这个疑问，deeocopy的话，self.model的参数就变了，就不是θ，而是θ‘了，这和文章中伪代码的Line 11 有出入。 另外，我发现表2 (M2): Remove parameter generator的实验结果和表1的MAML一样，难道MAML==(M2): Remove parameter generator，据我所知，MAML应该是没有graph reconstruction loss的吧

我也有相同的疑问，149行deepcopy的话，是否相当于没有使用MAML的思想，只是扩大了数据量统一地在support集和query集上迭代地更新参数。请问如果要按照MAML的思想应该改如何修改代码呢？我目前尝试了2种方式。（以下initial_model的参数即为初始的θ）
1）我尝试在更新θ之前deepcopy最初始的模型参数，但这样梯度无法回传
```
self.net = deepcopy(init_model) 
self.meta_optim.zero_grad() 
loss_q.backward() 
self.meta_optim.step()
```
2）在进行meta update之前，使用load_state_dict将模型参数手动设置为初始的参数θ：
```
new_paras = OrderedDict(zip(self.net.state_dict().keys(), list(init_model.state_dict().values())))
self.net.load_state_dict(new_paras) 
self.meta_optim.zero_grad() 
loss_q.backward() 
self.meta_optim.step()
```
但是这样出现了一个错误：
`one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [128, 2]], which is output 0 of AsStridedBackward0, is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).`

**Comment by RobinLu1209** [comment]
感谢大家对这一问题的指出，我们会进行修正。当前代码可以视为是对我们方法的一个训练策略的简化版本。
